src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -197,47 +239,111 @@ def get_cloud(cloud_name):
     return cloud
 
 
-def add_cloud(cloud):
-    if _get_cloud(cloud.name):
-        raise CloudAlreadyRegisteredException(cloud.name)
+def get_active_cloud():
+    return get_cloud(get_active_cloud_name())
+
+
+def get_cloud_subscription(cloud_name):
+    config = configparser.SafeConfigParser()
+    config.read(CLOUD_CONFIG_FILE)
+    try:
+        return config.get(cloud_name, 'subscription')
+    except (configparser.NoOptionError, configparser.NoSectionError):
+        return None
+
+
+def set_cloud_subscription(cloud_name, subscription):
+    if not _get_cloud(cloud_name):
+        raise CloudNotRegisteredException(cloud_name)
+    config = configparser.SafeConfigParser()
+    config.read(CLOUD_CONFIG_FILE)
+    if subscription:
+        config.set(cloud_name, 'subscription', subscription)
+    else:
+        config.remove_option(cloud_name, 'subscription')
+    if not os.path.isdir(GLOBAL_CONFIG_DIR):
+        os.makedirs(GLOBAL_CONFIG_DIR)
+    with open(CLOUD_CONFIG_FILE, 'w') as configfile:
+        config.write(configfile)
+
+
+def _set_active_subscription(cloud_name):
+    from azure.cli.core._profile import (Profile, _ENVIRONMENT_NAME, _SUBSCRIPTION_ID,
+                                         _STATE, _SUBSCRIPTION_NAME)
+    profile = Profile()
+    subscription_to_use = get_cloud_subscription(cloud_name) or \
+                          next((s[_SUBSCRIPTION_ID] for s in profile.load_cached_subscriptions()  # noqa # pylint: disable=line-too-long
+                                if s[_STATE] == 'Enabled'),
+                               None)
+    if subscription_to_use:
+        try:
+            profile.set_active_subscription(subscription_to_use)
+            sub = profile.get_subscription(subscription_to_use)
+            logger.warning("Active subscription switched to '%s (%s)'.",
+                           sub[_SUBSCRIPTION_NAME], sub[_SUBSCRIPTION_ID])
+        except CLIError as e:
+            logger.error(e)
```

Comment:
This catches the "The subscription of '...' does not exist or has more than one match" error.
I don't want that to fail the whole command because of this.
However, I'll change this to warning instead of error.